src/react/components/pages/editorPage/canvas.test.tsx

```diff
@@ -15,37 +15,41 @@ import { RegionsManager } from "vott-ct/lib/js/CanvasTools/Region/RegionsManager
 import { SelectionMode, AreaSelector } from "vott-ct/lib/js/CanvasTools/Selection/AreaSelector";
 
 describe("Editor Canvas", () => {
-    let wrapper: ReactWrapper<ICanvasProps, ICanvasState, Canvas> = null;
-    const onAssetMetadataChanged = jest.fn();
 
     function createTestRegionData() {
         const testRegionData = new RegionData(0, 0, 100, 100,
             [new Point2D(0, 0), new Point2D(1, 0), new Point2D(0, 1), new Point2D(1, 1)], RegionDataType.Rect);
         return testRegionData;
     }
 
-    function createComponent(canvasProps: ICanvasProps, assetPreviewProps: IAssetPreviewProps)
+    function createComponent(canvasProps?: ICanvasProps, assetPreviewProps?: IAssetPreviewProps)
         : ReactWrapper<ICanvasProps, ICanvasState, Canvas> {
-        return mount(
-            <Canvas {...canvasProps}>
-                <AssetPreview {...assetPreviewProps} />
-            </Canvas>,
-        );
+            const props = createProps();
+            const cProps = canvasProps || props.canvas;
+            const aProps = assetPreviewProps || props.assetPreview;
+            return mount(
+                <Canvas {...cProps}>
+                    <AssetPreview {...aProps} />
+                </Canvas>,
+            );
+    }
+    function getAssetMetadata() {
+        return MockFactory.createTestAssetMetadata(
+            MockFactory.createTestAsset(), MockFactory.createTestRegions());
     }
 
     function createProps() {
-        const assetMetadata = MockFactory.createTestAssetMetadata(MockFactory.createTestAsset("test"));
 
         const canvasProps: ICanvasProps = {
-            selectedAsset: assetMetadata,
-            onAssetMetadataChanged,
+            selectedAsset: getAssetMetadata(),
+            onAssetMetadataChanged: jest.fn(),
             editorMode: EditorMode.Rectangle,
             selectionMode: SelectionMode.RECT,
             project: MockFactory.createTestProject(),
         };
 
         const assetPreviewProps: IAssetPreviewProps = {
-            asset: assetMetadata.asset,
+            asset: getAssetMetadata().asset,
```

Comment:
Why in 44 can you just call "getAssetMetadata()" whereas here you use "getAssetMetadata().asset"?

Reply:
Kind of a weird naming scheme... The `.asset` property refers to the actual `IAsset` while this object is the `IAssetMetadata`